riscv/decode.h

```diff
@@ -3038,6 +3038,21 @@ reg_t index[P.VU.vlmax]; \
     WRITE_RD(sext_xlen(rd)); \
   }
 
+#define ZICBOM_OP(field, BODY) \
+  require_extension(EXT_ZICBOM); \
+  reg_t m##field = get_field(STATE.menvcfg->read(), MENVCFG_##field); \
+  reg_t s##field = get_field(STATE.senvcfg->read(), SENVCFG_##field); \
+  reg_t h##field = get_field(STATE.henvcfg->read(), HENVCFG_##field); \
+  if (((STATE.prv != PRV_M) && (m##field == 0)) || \
+      ((STATE.prv == PRV_U && !STATE.v) && (s##field == 0))) \
+    throw trap_illegal_instruction(insn.bits()); \
+  else if (STATE.v && ((h##field == 0) || \
+                       ((STATE.prv == PRV_U) && (s##field == 0)))) \
+    throw trap_virtual_instruction(insn.bits()); \
+  else { \
+    BODY \
```

Comment:
There is no need for this `else` and hence no need for `BODY` to be a macro op. It would be cleaner to have this work like the other `require_*` macros. Something like (for `cbo.clean`):
```
require_zicbom(CFCFE);
MMU.clean_inval(RS1, true, false);
```

Reply:
The reason why I use ZICBOM_OP intead of require_zicbom is that we need reuse the varaible m/s/h##field in cbo_inval.h, but require_* seems to be a separate block for check.

Reply:
together with cbo.zero. May be we can change to:
```
require_extension(EXT_ZICBOM);
DECLARE_XENVCFG(CFCFE);
require_envcfg(CFCFE);
MMU.clean_inval(RS1, true, false);
```

Reply:
Yuck...ok never mind.

But perhaps you should indent the lines inside the `{}` in the calls to `ZICBOM_OP`.